scality_manila_utils/utils.py

```diff
@@ -0,0 +1,177 @@
+    :raises: :py:class:`scality_manila_utils.exceptions.EnvironmentException`
+        if the binary couldn't be found
+    """
+    for path in paths:
+        if os.path.exists(os.path.join(path, binary)):
+            return
+
+    log.error("No '%s' found in PATH (%s)", binary, ', '.join(paths))
+    raise EnvironmentException("Unable to find '{0:s}', make sure it "
+                               "is installed".format(binary))
+
+
+def process_check(process):
+    """
+    Check if a process is running.
+
+    :param process: process name
+    :type process: string
+    :raises: :py:class:`scality_manila_utils.exceptions.EnvironmentException`
+        if the process isn't running
+    """
+    process_pids = find_pids(process)
+    if not process_pids:
+        log.error("'%s' is not running", process)
+        raise EnvironmentException("Could not find '{0:s}' running, "
+                                   "make sure it is "
+                                   "started".format(process))
+
+
+def safe_write(text, path, permissions=0o644):
+    """
+    Write contents to file in a safe manner.
+
+    Write contents to a tempfile and then move it in place. This is guarenteed
+    to be atomic on a POSIX filesystem.
+
+    :param text: the content to write to file
+    :type text: string
+    :param path: path to write to
+    :type path: string
+    :param permissions: file permissions
+    :type permissions: int (octal)
+    """
+    # Make sure that the temporary file lives on the same fs
+    log.debug("Writing '%s'", path)
+    target_dir, _ = os.path.split(path)
+    with tempfile.NamedTemporaryFile(mode='wt', dir=target_dir,
+                                     delete=False) as f:
+        os.chmod(f.name, permissions)
+        f.write(text)
+        os.fsync(f.fileno())
+        os.rename(f.name, path)
+
+    # fsync the directory holding the file just written and moved
+    dir_fd = None
+    try:
+        dir_fd = os.open(target_dir, os.O_RDONLY)
+        os.fsync(dir_fd)
+    finally:
+        if dir_fd is not None:
+            os.close(dir_fd)
+
+
+@contextlib.contextmanager
+def nfs_mount(export_path):
+    """
+    Mount an NFS filesystem, and keep it mounted while in context.
+
+    :param export_path: exported filesystem to mount, eg. `127.0.0.1:/`
+    :type export_path: string
+    :returns: path to where the filesystem was mounted
+    """
+    try:
+        mount_point = tempfile.mkdtemp()
+        subprocess.check_call(['mount', export_path, mount_point])
+        log.debug("Mounted nfs root '%s' at '%s'", export_path, mount_point)
+    except (OSError, subprocess.CalledProcessError):
+        log.exception('Unable to mount NFS root')
+        raise
+
+    try:
+        yield mount_point
+
+    finally:
+        try:
+            subprocess.check_call(['umount', mount_point])
+            log.debug('Unmounted nfs root')
+            os.rmdir(mount_point)
+        except subprocess.CalledProcessError:
```

Comment:
Consider moving the `except` clauses more closely to their origin: one for the `umount` call, one for the `rmdir` call, both only handling the expected exceptions.